frontend/src/metabase/admin/datamodel/datamodel.js

```diff
@@ -257,164 +24,19 @@ export const FETCH_REVISIONS = "metabase/admin/datamodel/FETCH_REVISIONS";
-    },
-  },
-  [],
-);
-
-const editingDatabase = handleActions(
-  {
-    [INITIALIZE_METADATA]: { next: (state, { payload }) => payload.database },
-    [SELECT_DATABASE]: {
-      next: (state, { payload }) => (payload ? payload : state),
-    },
-    [RETIRE_SEGMENT]: { next: (state, { payload }) => payload },
-    [RETIRE_METRIC]: { next: (state, { payload }) => payload },
-  },
-  null,
-);
-
-const editingTable = handleActions(
-  {
-    [INITIALIZE_METADATA]: {
-      next: (state, { payload }) => payload.tableId || null,
-    },
-    [SELECT_TABLE]: { next: (state, { payload }) => payload },
-  },
-  null,
-);
-
-const segments = handleActions(
-  {
-    [GET_SEGMENT]: {
-      next: (state, { payload }) => ({
-        ...state,
-        [payload.id]: momentifyTimestamps(payload),
-      }),
-    },
-    [CREATE_SEGMENT]: {
-      next: (state, { payload }) => ({
-        ...state,
-        [payload.id]: momentifyTimestamps(payload),
-      }),
-    },
-    [UPDATE_SEGMENT]: {
-      next: (state, { payload }) => ({
-        ...state,
-        [payload.id]: momentifyTimestamps(payload),
-      }),
-    },
-    [DELETE_SEGMENT]: {
-      next: (state, { payload }) => {
-        state = { ...state };
-        delete state[payload.id];
-        return state;
-      },
-    },
-  },
-  {},
-);
-
-const metrics = handleActions(
-  {
-    [GET_METRIC]: {
-      next: (state, { payload }) => ({
-        ...state,
-        [payload.id]: momentifyTimestamps(payload),
-      }),
-    },
-    [CREATE_METRIC]: {
-      next: (state, { payload }) => ({
-        ...state,
-        [payload.id]: momentifyTimestamps(payload),
-      }),
-    },
-    [UPDATE_METRIC]: {
-      next: (state, { payload }) => ({
-        ...state,
-        [payload.id]: momentifyTimestamps(payload),
-      }),
-    },
-    [DELETE_METRIC]: {
-      next: (state, { payload }) => {
-        state = { ...state };
-        delete state[payload.id];
-        return state;
-      },
-    },
-  },
-  {},
-);
-
-const tableMetadata = handleActions(
-  {
-    [LOAD_TABLE_METADATA]: {
-      next: (state, { payload }) =>
-        payload && payload.table ? payload.table : null,
-      throw: (state, action) => null,
-    },
-  },
-  null,
-);
-
 const previewSummary = handleActions(
-  {
-    [UPDATE_PREVIEW_SUMMARY]: { next: (state, { payload }) => payload },
-  },
+  { [UPDATE_PREVIEW_SUMMARY]: { next: (state, { payload }) => payload } },
   null,
 );
 
-const revisionObject = handleActions(
-  {
-    [FETCH_REVISIONS]: {
-      next: (state, { payload: revisionObject }) => revisionObject,
-    },
-  },
+const revisions = handleActions(
+  { [FETCH_REVISIONS]: { next: (state, { payload }) => payload } },
   null,
 );
 
-export default combineReducers({
-  databases,
-  idfields,
-  editingDatabase,
-  editingTable,
-  segments,
-  metrics,
-  tableMetadata,
-  previewSummary,
-  revisionObject,
-});
+export default combineReducers({ previewSummary, revisions });
```

Comment:
I removed _almost_ everything in datamodel.js. The two remaining pieces are revisions for metrics and segments and the preview summary. Revisions felt like they could be part their own entity almost, and I'm not sure if there's a better place for the preview summary since it's used for UI in the data model. Any thoughts?